### PR TITLE
Minor fixes to array functions

### PR DESCRIPTION
**Pull request by emile-00**
## Info

- **Description**

  Address additional comments from https://github.com/risingwavelabs/risingwave-docs/pull/1217#pullrequestreview-1612319368

- **Notes**

  [ Include any supplementary context or references here. ]

- **Related code PR**

  [ Provide a link to the relevant code PR here, if applicable. ]

- **Related doc issue**
  
  Resolves [ Provide a link to the relevant doc issue here, if applicable. ]

## For reviewers

- **Preview**

  [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **upcoming** version of the documentation. ]

- **Key points**

  [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

**Review by xiangjinwu — COMMENTED** [review]
Content lgtm. But make sure the conflict is resolved properly.

File: docs/sql/functions-operators/sql-function-array.md [hunk 1]
```diff
@@ -6,6 +6,44 @@ title: Array functions
...
+array[66] || array[123] → {66, 123}
+
+array_cat(array[array[66]], array[233]) → {{66}, {233}}
+
+array_cat(array[233], array[array[66]]) → {{233}, {66}}
+```
+
+---
+
+### `array_dims`
```
Comment (xiangjinwu) [inline docs/sql/functions-operators/sql-function-array.md:45]: After conflict resolution, title is `array_dims` but content below is `array_append`.

Comment (hengm3467) [reply docs/sql/functions-operators/sql-function-array.md:45]: @xiangjinwu Thanks for spotting this. I just fixed it.

**Review by hengm3467 — APPROVED** [review]

File: docs/sql/functions-operators/sql-function-array.md [hunk 2]
```diff
@@ -40,15 +42,50 @@ array_cat(array[233], array[array[66]]) → {{233}, {66}}
...
+
+```bash title=Example
+array_dims ( array[2,3,4] ) → [1:3]
+```
+
+---
+
+### `array_cat`
+
+Concatenates two arrays with the same data type. The `||` operator can also be used.
```
Comment (xiangjinwu) [inline docs/sql/functions-operators/sql-function-array.md]: Seems duplicate

Comment (hengm3467) [reply docs/sql/functions-operators/sql-function-array.md]: Fixed. My bad not spotting this one.

**Review by xiangjinwu — APPROVED** [review]

**Pull request merged**